src/Tank.cpp

```diff
@@ -38,18 +43,46 @@ void TankPart::draw(sf::RenderWindow &window, const float x, const float y) {
 }
 
 Tank::Tank(float x, float y, sf::Texture &body, sf::Texture &tower, sf::Texture &shot,
-           std::unique_ptr<Engine> &&engine)
-    : mPos({x, y}), mBody(body), mTower(tower), mShot(shot), mEngine(std::move(engine)) {
-  mTower.set_rotation(180);
-  mShot.set_rotation(180);
+           sf::Texture &tracks, std::unique_ptr<Engine> &&engine,
+           const TracesHandlerConfig &traces_handler_config)
+    : mPos({x, y}),
+      mBody(body),
+      mTower(tower),
+      mShot(shot),
+      mEngine(std::move(engine)),
+      mTracesHandler(std::make_unique<TracesHandler>(tracks, mBody.get_sprite(), mPos,
+                                                     traces_handler_config)) {
+  set_rotation(180);
+  mBody.get_sprite().setPosition(mPos);
+  mTower.get_sprite().setPosition(mPos);
+  mShot.get_sprite().setPosition(mPos);
 }
 
 Tank::Tank(const Tank &rhs)
     : mPos(rhs.mPos),
+      mCurrentSpeed(rhs.mCurrentSpeed),
+      mShotStart(rhs.mShotStart),
+      mDrawShot(rhs.mDrawShot),
       mBody(rhs.mBody),
       mTower(rhs.mTower),
       mShot(rhs.mShot),
-      mEngine(rhs.mEngine->copy()) {}
+      mEngine(rhs.mEngine->copy()),
+      mTracesHandler(std::make_unique<TracesHandler>(rhs.mTracesHandler->get_trace_texture(),
+                                                     mBody.get_sprite(), mPos,
+                                                     rhs.mTracesHandler->get_config())) {}
+
+Tank::Tank(Tank &&rhs)
+    : mPos(std::move(rhs.mPos)),
+      mCurrentSpeed(std::move(rhs.mCurrentSpeed)),
+      mShotStart(std::move(rhs.mShotStart)),
+      mDrawShot(std::move(rhs.mDrawShot)),
+      mBody(std::move(rhs.mBody)),
+      mTower(std::move(rhs.mTower)),
+      mShot(std::move(rhs.mShot)),
+      mEngine(std::move(rhs.mEngine)),
+      mTracesHandler(std::make_unique<TracesHandler>(rhs.mTracesHandler->get_trace_texture(),
```

Comment:
I think we should implement move operators for TracesHandler

Reply:
`TracesHandler` depends on a reference to `sf::Sprite` (tank body), so when `Tank` is moved we have to create new `TracesHandler` instance that will be linked to new `sf::Sprite`. When `TracesHandler` had default move operators this caused hard to diagnose error, because implicit move for `Tank` did not work correctly (`TracesHandler` was using reference to a dangling object - old tank body).